jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -432,59 +561,89 @@ def vue_do_aper_phot(self, *args, **kwargs):
                         p_val = p_val.value
                     fit_tmp.append({'function': param, 'result': f'{p_val:.4e}'})
 
-            self.results = tmp
+        self.results = tmp
+        self.result_available = True
+
+        if update_plots:
             self.fit_results = fit_tmp
-            self.result_available = True
             self.plot_available = True
 
+        return phot_table, fit_model
+
+    def vue_do_aper_phot(self, *args, **kwargs):
+        if self.dataset_selected == '' or self.aperture_selected == '':
+            self.hub.broadcast(SnackbarMessage(
+                "No data for aperture photometry", color='error', sender=self))
+            return
+
+        try:
+            if self.multiselect:
+                # even though plots aren't show in the UI when in multiselect mode,
+                # we'll create the last entry so if multiselect is disabled, the last
+                # iteration will show and not result in confusing behavior

```

Comment:
Personally, I find it less confusing if you clear everything when someone toggle multi-select off. I don't want us to end up in the situation where there is a subtle bug and the result actually does not map to the options that show up when you turn off multi-select. Is there ever a situation where someone turns off multi-select but somehow find that one result worth keeping but just that one?

Reply:
I'm open to that idea, but it will take a little extra logic, so let's decide what path we're taking wrt the radial profile first and then decide if this is worth pursuing now or not.